template.js

```diff
@@ -9,71 +9,81 @@ const JSON = require('JSON');
 const returnResponse = require('returnResponse');
 const setCookie = require('setCookie');
-const templateDataStorage = require('templateDataStorage');
 const sendHttpGet = require('sendHttpGet');
-const getTimestampMillis = require('getTimestampMillis');
 const sendHttpRequest = require('sendHttpRequest');
 const setResponseBody = require('setResponseBody');
 const setResponseHeader = require('setResponseHeader');
 const setResponseStatus = require('setResponseStatus');
 const getContainerVersion = require('getContainerVersion');
 const getRemoteAddress = require('getRemoteAddress');
+const getType = require('getType');
 const path = getRequestPath();
 
-// Check if this Client should serve exponea.js file
-if (path === data.proxyJsFilePath) {
-    claimRequest();
+const cookieWhiteList = ['xnpe_' + data.projectToken, '__exponea_etc__', '__exponea_time2__'];
+const headerWhiteList = ['referer', 'user-agent', 'etag'];
 
-    const now = getTimestampMillis();
-    const thirty_minutes_ago = now - (30 * 60 * 1000);
 
-    if (templateDataStorage.getItemCopy('exponea_js') == null || templateDataStorage.getItemCopy('exponea_stored_at') < thirty_minutes_ago) {
-        sendHttpGet('https://api.exponea.com/js/exponea.min.js', {headers: {'X-Forwarded-For': getRemoteAddress()}}).then((result) => {
-            if (result.statusCode === 200) {
-                templateDataStorage.setItemCopy('exponea_js', result.body);
-                templateDataStorage.setItemCopy('exponea_headers', result.headers);
-                templateDataStorage.setItemCopy('exponea_stored_at', now);
-            }
-            sendProxyResponse(result.body, result.headers, result.statusCode);
-        });
-    } else {
-        sendProxyResponse(
-            templateDataStorage.getItemCopy('exponea_js'),
-            templateDataStorage.getItemCopy('exponea_headers'),
-            200
-        );
-    }
+// Check if this Client should serve exponea.js file
+if (splitAndSanitize(data.proxyJsFilePath).reduce((res,proxyPath)=>res || equalOrStartsOrEnds(path,proxyPath), false)) {
+    claimRequest();
+    log({
+        'Name': 'Exponea',
+        'Type': 'Serving JS',
+        'path': path
+    });
+    sendHttpGet(data.targetAPI+path, {headers: {'X-Forwarded-For': getRemoteAddress()}}).then((result) => {
+      sendProxyResponse(result.body, result.headers, result.statusCode);
+    });
 }
 
-// Check if this Client should serve exponea.js.map file (Just only to avoid annoying error in console)
-if (path === '/exponea.min.js.map') {
-    sendProxyResponse('{"version": 1, "mappings": "", "sources": [], "names": [], "file": ""}', {'Content-Type': 'application/json'}, 200);
-}
+var validPaths = [
+  '/bulk',
+  '/managed-tags/show',
+  '/campaigns/banners/show',
+  '/campaigns/experiments/show',
+  '/campaigns/html/get',
+  '/optimization/recommend/user',
+  '/webxp/projects/'+data.projectToken+'/',
+  '/webxp/data/modifications/'+data.projectToken+'/',
+  '/webxp/bandits/reward'
```

Comment:
That's an excellent idea for refactoring; however, this array should be spliced into two parts.
One that serves without caching and one with template caching.